synapse/storage/engines/postgres.py

```diff
@@ -15,16 +15,14 @@
 
 import logging
 
-from ._base import IncorrectDatabaseSetup
+from ._base import BaseDatabaseEngine, IncorrectDatabaseSetup
 
 logger = logging.getLogger(__name__)
 
 
-class PostgresEngine(object):
-    single_threaded = False
-
+class PostgresEngine(BaseDatabaseEngine):
```

Comment:
Isn't `BaseDatabaseEngine` missing a connection type? (seeing as `Sqlite3Engine` has `BaseDatabaseEngine[sqlite3.Connection]`)

Reply:
well, I don't think it's mandatory, and the problem is that there isn't an equivalent connection type for Postgres.

Reply:
Ah, fair enough then